glean-core/src/upload/mod.rs

```diff
@@ -140,6 +145,99 @@ impl PingUploadManager {
             None => PingUploadTask::Done,
         }
     }
+
+    /// Processes the response from an attempt to upload a ping.
+    ///
+    /// Spawns a thread for any disk IO necessary.
+    ///
+    /// Based on the HTTP status of said response,
+    /// the possible outcomes are:
+    ///
+    /// * **200 - 299 Success**
+    ///   Any status on the 2XX range is considered a succesful upload,
+    ///   which means the corresponding ping file can be deleted.
+    ///   _Known 2XX status:_
+    ///   * 200 - OK. Request accepted into the pipeline.
+    ///
+    /// * **400 - 499 Unrecoverable error**
+    ///   Any status on the 4XX range means something our client did is not correct.
+    ///   It is unlikely that the client is going to recover from this by retrying,
+    ///   so in this case the corresponding ping file can also be deleted.
+    ///   _Known 4XX status:_
+    ///   * 404 - not found - POST/PUT to an unknown namespace
+    ///   * 405 - wrong request type (anything other than POST/PUT)
+    ///   * 411 - missing content-length header
+    ///   * 413 - request body too large Note that if we have badly-behaved clients that
+    ///           retry on 4XX, we should send back 202 on body/path too long).
+    ///   * 414 - request path too long (See above)
+    ///
+    /// * **Any other error**
+    ///   For any other error, a warning is logged and the ping is re-enqueued.
+    ///   _Known other errors:_
+    ///   * 500 - internal error
+    ///
+    /// # Arguments
+    ///
+    /// `uuid` - The UUID of the ping in question.
+    /// `status` - The HTTP status of the response.
+    ///
+    /// # Return value
+    ///
+    /// Returns the JoinHandle for the spawned thread for test syncronization purposes.
```

Comment:
```suggestion
    /// Returns a handle for the spawned I/O thread. Callers can use this to synchronize on the I/O processing.
```

I'm not a fan of making a return value solely documented for a test purpose.
What do you think about the alternate phrasing?

Is a (non-test) caller supposed to do anything with the thread?

Reply:
It _can_ do something with the thread if it wants to. But for the uses I forsee for it, I don't think it will. I do prefer your phrasing though.